modules/programs/mpv.nix

```diff
@@ -49,30 +49,28 @@ let
   renderDefaultProfiles = profiles:
     renderOptions { profile = concatStringsSep "," profiles; };
 
-  mpvPackage = if cfg.scripts == [ ] then
-    pkgs.mpv
-  else
-    pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = cfg.scripts; };
-
```

Comment:
``` suggestion
 mpvPackage = if cfg.scripts == [ ] then
    cfg.package
  else
    pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = cfg.scripts; };
```

I would keep the `scripts` option and change the logic to allow the user to assign to `package` (by removing the `readOnly`).

Reply:
Look at the other answer. If this use case of setting script multiple times is really common I can do it, but I would prefer not to. This will complicate the module a lot since I also need to do an assertion (otherwise the user can do the wrong thing, like setting both scripts and a custom derivation).

Also, why scripts is so special that it needs its own option, while things like enabling optional options inside mpv derivation (like `vapoursynthSupport`) is not.

Reply:
I tried to do like this, but I didn't found a way to solve the infinity recursion in this part:

```nix
{
  programs.mpv.package = mpvPackage;
}
```

Without this I would break #1524 use case that seems less niche than the solution for supporting both `programs.mpv.scripts` and `programs.mpv.package`. So again I am in favor of the current solution of this PR, now because simply supporting both seems impossible.

Here are a pastebin (sorry for the lack of syntax highlighting): http://ix.io/2MwK

Reply:
Your pastebin looks good.
I would make the default value of `package` be `pkgs.mpv` instead of `null` considering the rest of the logic.

As for the package, we usually add an option like : 
```
finalPackage = mkOption {
  type = types.package;
  internal = true;
  readOnly = true;
};
# .... config block
config = {
  programs.mpv.finalPackage = mpvPackage;
};
```

This exposes the configured mpv package to be reused elsewhere.